tsdb/index/postings.go

```diff
@@ -79,6 +79,70 @@ func (p *MemPostings) SortedKeys() []labels.Label {
 	return keys
 }
 
+type HeadStats struct {
+	Name  string
+	Count uint64
+}
+
+type Stats struct {
+	CardinalityMetricsStats []HeadStats
+	CardinalityLabelStats   []HeadStats
+	LabelValueStats         []HeadStats
+	LabelValuePairsStats    []HeadStats
+}
+
+// Calculating Cardinality Stats
+func (p *MemPostings) CardinalityStats() *Stats {
+	metrics := make([]HeadStats, 0)
+	labels := make([]HeadStats, 0)
+	labelValueLenght := make([]HeadStats, 0)
+	labelValuePairs := make([]HeadStats, 0)
+	const maxNumOfRecords = 10
+	var size uint64
+	p.mtx.RLock()
+
+	for n, e := range p.m {
+		if n == "" {
+			continue
+		}
+		labels = append(labels, HeadStats{Name: n, Count: uint64(len(e))})
+		size = 0
+		if n == "__name__" {
```

Comment:
The TSDB shouldn't know that __name__ is special.

Reply:
Removed the ` __name__ ` from TSDB.